csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -240,6 +240,16 @@ module Public {
      */
     predicate isAutoGenerated() { none() }
   }
+
+  /** A callable with a flow summary stating there is no flow via the callable. */
+  class NegativeSummarizedCallable extends SummarizedCallableBase {
+    NegativeSummarizedCallable() { negativeSummaryElement(this, _) }
```

Comment:
I did not expect to see negative summaries as part of `SummarizedCallableBase`. That seems wrong to me.

Reply:
Not exactly sure, what you mean.
Maybe it is the naming of the class SummarizedCallableBase that is misleading - maybe SummarizedCallableBase should be renamed to something like SyntheticCallable(Base). Note that summarized callables (SummarizedCallable) extends SummarizedCallableBase.

Reply:
But negative summaries don't need data flow node synthesis. The `SummarizedCallable` class is the set of callables for which we synthesise a number of nodes to enable flow-through.

Reply:
Not exactly sure how that could happen.
Note that there are three classes `SummarizedCallable`, `SummarizedCallableBase` and `NegativeSummarizedCallable`. SummarizedCallable is an abstract class that extends `SummarizedCallableBase` and `NegativeSummarizedCallable` extends `SummarizedCallableBase` and not `SummarizedCallable`.
Should we change the same of `SummarizedCallableBase` to something else?

Reply:
Never mind. This was me misreading and thinking that `SummarizedCallableBase` was an abstract class.